### DISCUSSION TITLE
curl state transition log is disappeared

### DISCUSSION THREAD

**Discussion by soulArche** [curl/curl#11404] [Q&A] [answered]
I've seen below logs when curl connect state is changed.

STATE: INIT => CONNECT handle 0x558b71cb00; line 1835 (connection #-5000)
STATE: CONNECT => CONNECTING handle 0x558b71cb00; line 1896 (connection #56)
STATE: CONNECTING => PROTOCONNECT handle 0x558b71cb00; line 2028 (connection #56)
STATE: PROTOCONNECTING => DO handle 0x558b71cb00; line 2067 (connection #56)
STATE: DO => DID handle 0x558b71cb00; line 2147 (connection #56)
STATE: DID => PERFORMING handle 0x558b71cb00; line 2266 (connection #56)
STATE: PERFORMING => DONE handle 0x558b71cb00; line 2465 (connection #56)

Those logs are very helpful to debug each curl connection state.
However I couldn't seen the state transition logs anymore with updated curl version(8.1.0).
When was the debug log disable? Will it be enabled in the future?

**Comment by dfandrich** [comment, marked as answer]
These logs are only available when curl has been compiled with debug mode enabled.

**Comment by soulArche** [comment]
I've added the related link about enabling debug mode and potential issue.
https://github.com/curl/curl/issues/10269
https://github.com/curl/curl/pull/10278
https://curl.se/mail/lib-2023-01/0039.html
